### PR TITLE
Auto ticket link

### PR DESCRIPTION
**Pull request by seanim0920**
### Problem
It was too inconvenient linking trello tickets to github PR's. We can make the process more convenient, giving devs an easier time keeping the trello board up to date (which will in turn keep the daily summary up to date).

Previous workflow:
1. Open trello
2. Write ticket
3. Make a new branch
4. Push the branch
5. Open github
6. Create PR
7. Copy PR link
8. Open trello
9. Find corresponding ticket
10. Attach PR to ticket

### Solution
* Add the trello card id in the branch name or PR description. Then add scripts to the pull request process to automatically link the PR to trello and vice versa when the PR is opened.

### Implementation
* Add script "npm run pr" to open pr form and fill in title/description based on card id from the branch name or params.
* Add github action to attach the newly opened pr to the trello card.

Notes: Trello api key and Trello api token are needed in the .env.infra in order for the "npm run pr" script to work. These are in the slack channel.

Now when opening a PR, the corresponding trello ticket gets linked in one fell swoop.

-----------------------------------

New standard workflow:
1. Open trello
2. Write ticket
3. Copy ticket ID
4. Make a new branch with the ticket ID in the branch name
5. Push the branch
6. Run "npm run pr"
7. Create PR

Alternative workflow (making PR first):
1. Make a new branch
2. Push the branch
3. Open trello
4. Write ticket/Find matching ticket
5. Copy ticket ID
6. Run "npm run pr [ticketID]"
7. Create PR

-----------------------------------

This workflow should be reusable in the front-end and back-end repos.

### Other
* Added .env.template
* Needed to add "type":"module" to the package.json and enable .ts imports for the npm script to run as a .ts file
* Added .npmignore so .test.ts files and the npm-scripts aren't included in the package. For local development we can use a symlink to access the npm scripts and tests from other projects. Instructions are added to the readme.
* Changed package.json name to match repo name. The symlink is case sensitive so running "npm link tifshared" won't work.

TASK_y6zY4kQW

### CODE REVIEW

**Comment by tifapp** [comment]
Linked Ticket: https://trello.com/c/y6zY4kQW

**Comment by seanim0920** [comment]
Check that scripts arent included in bundle

**Comment by seanim0920** [comment]
Solved using symlinks

**Comment by seanim0920** [comment]
[github-trello integration](https://trello.com/c/y6zY4kQW/545-github-trello-integration)

File: npm-scripts/auto-pr-util.test.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    it('extracts and returns the ticket ID from the branch name when no ticket ID is provided', async () => {
+      const currentBranch = 'feature/TASK_456-implement-y';
+  
+      expect(getTicketId(currentBranch)).toBe('TASK_456');
+    });
+  
+    it('returns undefined and logs appropriate message when no ticket ID is found and none is provided', async () => {
+      const currentBranch = 'feature/implement-z';
+      expect(getTicketId(currentBranch)).toBeUndefined();
+    });
```
Comment (mhayes853) [inline npm-scripts/auto-pr-util.test.ts:26]: Minor Nit: This doesn't seem to test that a message was logged directly. There are 2 ways you can achieve this:
1. Override `console.log` and assert it was called a certain way.
2. Use the logging library in this repo, add a test `LogHandler` that's a jest mock, and assert the mock was called with a specific log message.

**Review by mhayes853 — APPROVED** [review]

File: npm-scripts/auto-pr-util.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import { Logger } from "logging";
+
+export const getRawIdFromTicketId = (ticketId: string) => {
+    const match = ticketId.match(/(?<=task_)\w+/i);
+    return match ? match[0] : null;
+};
+
+export const getTicketId = (logger: Logger, currentBranchName: string, ticketId?: string) => {
```
Comment (mhayes853) [inline npm-scripts/auto-pr-util.ts:8]: You don't need the logger as a parameter just for testing. You can create a file-private logger in the global scope and use it directly in `getTicketId`. For testing, all you need to do is something like:
```ts
// In some test
const handler = jest.fn()
addLogHandler(handler)
getTicketId("i am an invalid ticket")
expect(handler).toHaveBeenCalledWith(...)
```
`Logger` takes care of logging to all available `LogHandler`s.

Comment (seanim0920) [reply npm-scripts/auto-pr-util.ts:8]: Thanks

**Review by mhayes853 — APPROVED** [review]

**Pull request merged**